src/interface_http.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+logger = logging.getLogger()
+
+from ops.framework import (
+    EventSource,
+    Object,
+    ObjectEvents,
+    StoredState
+)
+from ops.charm import RelationEvent
+from adapters.framework import FrameworkAdapter
+
+
+class PrometheusNewClientEvent(RelationEvent):
+    pass
+
+
+class PrometheusEvents(ObjectEvents):
+    new_client = EventSource(PrometheusNewClientEvent)
+
+
+class PrometheusInterface(Object):
+    on = PrometheusEvents()
+    state = StoredState()
+
+    def __init__(self, charm, relation_name):
+        super().__init__(charm, relation_name)
+        self.fw = FrameworkAdapter(self.framework)
+        self.relation_name = relation_name
+        self.fw.observe(charm.on[relation_name].relation_joined,
+                        self.on_relation_joined)
+        self.state.set_default(apps=[])
+
+    def render_relation_data(self):
+        logging.debug('render-relation-data in')
+        for relation in self.model.relations[self.relation_name]:
+            relation.data[self.model.unit]['prometheus-port'] = \
```

Comment:
I'm a little concerned with the server side being responsible for configuring the client side of the interface. This can potentially result in cascading errors which [Mark Shuttleworth has acknowledged is a problem](https://discourse.juju.is/t/critical-review-of-juju-today/3030/7?u=mmaglana). To avoid this, for interfaces, I think we should stick to just returning data from one side to another and leave the configuration logic to the charms themselves.

Reply:
Oh wait. Was I being a dumbass? This line sets the `prometheus-port` data that this part in Grafana consumes, huh? https://github.com/relaxdiego/charm-k8s-grafana/pull/5/files#diff-9b211c64f1736c7647297f7f1f3dcbbdR99

Reply:
Well, the behaviour has not changed - in the original charm version Grafana was configuring its datasource with parsing of Prometheus k8s pod spec; now it's grabbing the cluster-ip and port not from the spec, but from the relation. But the original idea is the same - we're getting the data about the remote (Prometheus) side and rendering the Grafana datasource config. What do you think?

Reply:
Yes this part totally makes sense.